Java/syntax_test_java.java

```diff
@@ -746,6 +746,67 @@ int operators() {
   }
 //^ meta.method.java meta.method.body.java punctuation.section.block.end.java
 
+  int numbers() {
+    a = 0 + 0L;
+//      ^ constant.numeric
+//          ^^ constant.numeric
+
+    a = 0xABCD + 0xAB_CD;
+//      ^^^^^^ constant.numeric
+//               ^^^^^^ constant.numeric
+
+    a = 0x_AB_CD - 0xAB_CD_;
+//      ^^^^^^^^ -constant.numeric
```

Comment:
Instead of not applying constant.numeric can you mark the _ as invalid.illegal or mark the whole thing as invalid

Reply:
I would say go for a compromise - scope everything except the trailing underscore as a number, and only scope the trailing underscore as invalid if followed immediately by a semicolon - that should avoid flickering colors.